qhana_plugin_runner/db/models/mutable_json.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2021 QHAna plugin runner contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""This module contains helper class to track changes in json columns."""
+
+from sqlalchemy import event, inspect
+from sqlalchemy.ext.mutable import Mutable
+from sqlalchemy.types import JSON
+from sqlalchemy_json import NestedMutableDict, NestedMutableList
+
+
+class NestedMutable(Mutable):
+    """SQLAlchemy `mutable` extension with nested change tracking."""
+
+    @classmethod
+    def _listen_on_attribute(cls, attribute, coerce, parent_cls):
+        """Overwrite method of base class in order to accept primitive (immutable) types. 
+        
+        Code is identical to MutableBase._listen_on_attribute apart from handling ``str``, ``float`` and ``int`` (which includes ``bool`` as ``bool`` is a subclass of ``int``)."""
```

Comment:
Check if the license of the copied / modified code is compatible with the Apache 2.0 license. If that's the case, satisfy the conditions of the license e.g. include the license notice and permission notice.